### DISCUSSION TITLE
Mutliple manifests artifacts in one OCIRepository

### DISCUSSION THREAD

**Discussion by vterdunov** [fluxcd/flux2#4113] [Q&A] [answered]
https://fluxcd.io/flux/components/source/ocirepositories/#layer-selector says I can use other media types that `flux push artifact` produces.

I want to run [pre deploy jobs](https://fluxcd.io/flux/use-cases/running-jobs/) (database migrations).
I want to try to put pre-deploy and deploy manifests in one image.

Questions:
- is this ok in general?  Because CI jobs that produces OCI Images may fall right after `pre-deploy` artifacts tagged as [latest](https://fluxcd.io/flux/cheatsheets/oci-artifacts/#deploy-latest-on-staging) that leads to `pre-deploy` jobs will be executed while `deploy` - not.
- Is there any tool/solution how to build OCI Image with custom media type with compatible with Flux.
Let's say `application/deployment.content.v1.tar+gzip`and `application/pre-deployment.content.v1.tar+gzip`

**Comment by stefanprodan** [comment, marked as answer]
> I want to try to put pre-deploy and deploy manifests in one image.

You can do this by having both directories in the same artifact. The Flux Kustomizations will use the same artifact but with different paths.
